jenkins/run-server-func-tests

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+export EXTRA_PODMAN_SWITCHES=${EXTRA_PODMAN_SWITCHES:-"--pull=always -e=COV_REPORT_XML=yes"}
+export IMAGE_KIND=${IMAGE_KIND:-"fedora"}
+export IMAGE_REPO=${IMAGE_REPO:-"quay.io/pbench"}
+export IMAGE_ROLE=${IMAGE_ROLE:-"ci"}
+
+export PB_CONTAINER_REG=${PB_CONTAINER_REG:-"images.paas.redhat.com"}
+export PB_REPO_NAME=${PB_REPO_NAME:-"pbench"}
+
+export PB_ELASTICSEARCH_IMAGE_NAME=${PB_ELASTICSEARCH_IMAGE_NAME:-"elasticsearch"}
+export PB_ELASTICSEARCH_IMAGE_TAG=${PB_ELASTICSEARCH_IMAGE_TAG:-"7.13.3"}
+export PB_POSTGRESQL_IMAGE_NAME=${PB_POSTGRESQL_IMAGE_NAME:-"postgresql-13"}
+export PB_POSTGRESQL_IMAGE_TAG=${PB_POSTGRESQL_IMAGE_TAG:-"latest"}
+export PB_SERVER_IMAGE_NAME=${PB_SERVER_IMAGE_NAME:-"pbench-server-ci"}
+export PB_SERVER_IMAGE_TAG=${PB_SERVER_IMAGE_TAG:-"main"}
+export PB_POD_NAME=${PB_POD_NAME:-"pbench-server_${PB_SERVER_IMAGE_TAG}"}
+
+export PB_COMMON_IMAGE_PULL_POLICY=${PB_COMMON_IMAGE_PULL_POLICY:-"Always"}
+export PB_ELASTICSEARCH_IMAGE_PULL_POLICY="${PB_ELASTICSEARCH_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
+export PB_POSTGRESQL_IMAGE_PULL_POLICY="${PB_POSTGRESQL_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
+export PB_SERVER_IMAGE_PULL_POLICY="${PB_SERVER_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
+
+export PB_DASHBOARD_DIR="${PB_DASHBOARD_DIR:-${PWD}/dashboard/build/}"
+
+function cleanup {
+    podman pod rm --force --ignore ${1} || true
+    podman container prune -f || true
+    podman container cleanup --all --rm || true
+    podman image prune -f
+}
+
+if [[ ${1} == "--cleanup" ]]; then
+    trap "cleanup ${PB_POD_NAME}" INT QUIT ERR EXIT
+fi
+
+envsubst < server/pbenchinacan/pod.yml | podman play kube -
+
+curl --silent localhost:8080/api/v1/endpoints >/dev/null
+while [[ ${?} -ne 0 ]]; do
+    sleep 1
+    echo "Checking pod is live ..."
+    curl --silent localhost:8080/api/v1/endpoints >/dev/null
+done
```

Comment:
Alternatively,
```
while [[ curl --silent localhost:8080/api/v1/endpoints >/dev/null ]]; do
    echo "Waiting for pod to respond...."
    sleep 1
done
```

Reply:
Good suggestion, though I had to remove the surrounding brackets.

Reply:
Thanks, and sorry (the brackets seemed to work OK in my testing, but it's even better if it works without them!).

I think there are many other places we should consider this approach -- I think we use `${?}` too much, and things could be simpler if we evaluated the command result directly.  (Yes, there is definitely a tradeoff, but...there are cases like this one where the value is clear, and I think there are many others where it is a win as well.)

Reply:
See PR #3112.